lib/ronin/cli/commands/compress.rb

```diff
@@ -0,0 +1,102 @@
+# frozen_string_literal: true
+#
+# Copyright (c) 2006-2023 Hal Brodigan (postmodern.mod3 at gmail.com)
+#
+# Ronin is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ronin is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ronin.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+require 'ronin/cli/string_methods_command'
+require 'ronin/support/compression'
+require 'ronin/support/archive'
+
+module Ronin
+  class CLI
+    module Commands
+      #
+      # Compress the files.
+      #
+      # ## Usage
+      #
+      #     ronin compress [option] [FILE ...]
+      #
+      # ## Options
+      #
+      #     -z, --zip                        zip archive the data
+      #     -t, --tar                        tar archive the data
+      #     -g, --gzip                       gzip compresses the data
+      #
+      # ## Arguments
+      #
+      #     [FILE ...]                       Optional file(s) to compress
+      #
+      class Compress < StringMethodsCommand
+        usage '[option] [FILE ...]'
+
+        option :gzip, short: '-g',
+                      desc:  'gzip compresses the data' do
+                        @format = :gzip
+                      end
```

Comment:
If we're going to support `--gzip`, we should also support a `--zlib` option. `ronin-support` also happens to have a `Ronin::Support::Compression::Zlib` module we can use.